File: WideBuy - FrontEnd/src/app/common/purchase/purchase.ts
import { Address } from '../address/address';
import { Customer } from '../customer/customer';
import { Order } from '../order/order';
import { OrderItem } from '../order-item/order-item';

export class Purchase {
  customer!: Customer;
  shippingAddress!: Address;
  order!: Order;
  orderItems!: OrderItem[];
}

File: WideBuy - BackEnd/src/main/java/com/kachur/WideBuy/dto/Purchase.java
package com.kachur.WideBuy.dto;

import com.kachur.WideBuy.entity.*;
import lombok.Data;

import java.util.Set;

@Data
public class Purchase {
    private Customer customer;
    private Address shippingAddress;
    private Order order;
    private Set<OrderItem> orderItems;
}